src/main/scala/scalismo/geometry/Coordinate.scala

```diff
@@ -41,3 +43,24 @@ abstract class Coordinate[D <: Dim: NDSpace, @specialized(Int, Float, Double) S]
   override def toString = data.deep.toString()
 }
 
+trait CoordinateOps[D <: Dim, Scalar, Repr <: Coordinate[D, Scalar]] { self: Coordinate[D, Scalar] =>
+
+  implicit val classTagScalar: ClassTag[Scalar]
+
+  protected def createRepr(data: Array[Scalar]): Repr
+
+  def mapWithIndex(f: (Scalar, Int) => Scalar): Repr = {
```

Comment:
check if utils.ArrayUtils.fastMap can be used, the array mapping is already implemented there.